### PR TITLE
Feature/schema based on diagram

### PR DESCRIPTION
**Pull request by lucy-sees**
In this part of the project, I worked with @sagieramos to do the following tasks:

- [x] Created a file named schema_based_on_diagram.sql where we implement the database from the diagram.
- [x] Created Joined table from many-to-many relationship which did not appear in the diagram.
- [x] Added the FK indexes.
- [X] Index all the FK

 
Screenshot from 2023-11-01 14-49-05

### CODE REVIEW

File: schema_based_on_diagram.sql [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+    treatment_id INT,
+    FOREIGN KEY (invoice_id) REFERENCES invoices(id),
+    FOREIGN KEY (treatment_id) REFERENCES treatments(id)
+);
+
+CREATE INDEX invoice_id_index ON invoice_items(invoice_id);
+CREATE INDEX treatment_id_index ON invoice_items(treatment_id);
+CREATE INDEX medical_history_id_index ON invoices (medical_history_id);
+CREATE INDEX patient_id_index ON medical_histories (patient_id);
+CREATE INDEX treatments_id_index ON medical_histories_treatments (treatments_id);
```
Comment (nahnahdev) [inline schema_based_on_diagram.sql:58]: - [ ] Great job so far @lucy-sees and @sagieramos There are a few changes I will suggest you implement and you'll be on your way to the next Milestone :wink: 
- [ ] I noticed here that, you have `created an index` for a `foreign key` on a table `medical_histories_treatments` that doesn't exist in your `schema_based_on_diagram.sql` file. So kindly remove it as it currently has no use being there :+1:

File: schema_based_on_diagram.sql [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+CREATE TABLE invoice_items(
+    id INT GENERATED ALWAYS AS IDENTITY PRIMARY KEY,
+    unit_price DECIMAL,
+    quantity INT,
+    total_price DECIMAL,
+    invoice_id INT,
+    treatment_id INT,
+    FOREIGN KEY (invoice_id) REFERENCES invoices(id),
+    FOREIGN KEY (treatment_id) REFERENCES treatments(id)
+);
```
Comment (nahnahdev) [inline schema_based_on_diagram.sql:43]: - [ ] Great job creating all the tables present on the diagram. However, you're currently missing one table which wasn't part of the diagram but needs to be created. This is a `"join table"` for the `many-to-many relationship` between `medical_histories` and `treatments` :+1: 
- [ ] So kindly ensure that that `join table` is created and all FOREIGN KEYS are set with INDEXES created for those Foreign Keys as well :wink:

**Review by nahnahdev — CHANGES_REQUESTED** [review]
## Project Status: CHANGES REQUIRED :recycle: 
Hi @lucy-sees and @sagieramos , 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!


## Highlights :fire: 
- The new tables were created in `schema_based_on_diagram.sql` file :heavy_check_mark: 
- All tables from the diagram are created using the `CREATE TABLE` command :heavy_check_mark: 
- Tables have all required columns with the correct type :heavy_check_mark: 
- All tables have a PRIMARY KEY set :heavy_check_mark: 
- All FOREIGN KEYS are set by using `refrences table_name(id)` in column definition :heavy_check_mark: 
- All FOREIGN KEYS have indexes created :heavy_check_mark: 

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better._

You can also consider:
- N/A

Cheers, and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by TadesseDev — APPROVED** [review]
### Hi @lucy-sees 

### I'm Tadesse Alemayehu your reviewer for this PR. :hand: 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 
AmazingCreepyFrogmouth-max-1mb




## Highlights
 - [x] Good job making the changes requested by the previous code reviewer. ✅


## Optional suggestions
- N/A

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**